src/Sylius/Component/Core/Promotion/Action/ShippingDiscountAction.php

```diff
@@ -43,8 +67,44 @@ public function execute(PromotionSubjectInterface $subject, array $configuration
     /**
      * {@inheritdoc}
      */
+    public function revert(PromotionSubjectInterface $subject, array $configuration, PromotionInterface $promotion)
+    {
+        if (!$subject instanceof OrderInterface && !$subject instanceof OrderItemInterface) {
```

Comment:
Asserts? I have one problem with this library. IIRC it can only throws invalid argument exceptions...